From b6b5888dfd7df290cf7216c656453eca0c2c24c6 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Wed, 21 Oct 2015 00:26:21 +0800
Subject: [PATCH] MLK-11741 cpufreq: imx: fix a potential issue in the imx7
 cpufreq driver

The 'OR' operation should be '||', so fix it.

Signed-off-by: Bai Ping <b51503@freescale.com>
(cherry picked from commit 9d592bb4453fc1769cff503b1533d0124a9a60bc)
---
 drivers/cpufreq/imx7-cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cpufreq/imx7-cpufreq.c b/drivers/cpufreq/imx7-cpufreq.c
index 28d4cdd..a620cf0 100644
--- a/drivers/cpufreq/imx7-cpufreq.c
+++ b/drivers/cpufreq/imx7-cpufreq.c
@@ -181,7 +181,7 @@ static int imx7d_cpufreq_probe(struct platform_device *pdev)
 	pll_arm = devm_clk_get(cpu_dev, "pll_arm");
 	pll_sys_main = devm_clk_get(cpu_dev, "pll_sys_main");
 
-	if (IS_ERR(arm_clk) | IS_ERR(arm_src) | IS_ERR(pll_arm) |
+	if (IS_ERR(arm_clk) || IS_ERR(arm_src) || IS_ERR(pll_arm) ||
 	    IS_ERR(pll_sys_main)) {
 		dev_err(cpu_dev, "failed to get clocks\n");
 		ret = -ENOENT;
-- 
1.8.0

